pkg/minikube/node/start.go

```diff
@@ -180,12 +181,40 @@ func Start(starter Starter, apiServer bool) (*kubeconfig.Settings, error) {
 		}
 	}
 
+	if err := restartKubeletIfNeeded(starter); err != nil {
+		return nil, errors.Wrap(err, "restarting kubelet")
+	}
+
 	wg.Wait()
 
 	// Write enabled addons to the config before completion
 	return kcs, config.Write(viper.GetString(config.ProfileName), starter.Cfg)
 }
 
+func restartKubeletIfNeeded(starter Starter) error {
+	if !starter.PreExists {
+		return nil
+	}
+	if !driver.IsKIC(starter.Host.DriverName) {
```

Comment:
Can you give more detail on why we think this is only a KIC issue? It seems like it should be a problem for any case which changes IP or configuration.

Reply:
fair, new changes have removed the kic-specification